source/grid/grid_tools.cc

```diff
@@ -2140,8 +2140,10 @@ namespace GridTools
     const unsigned int n_cpu = Utilities::MPI::n_mpi_processes(triangulation.get_communicator());
     std::vector<types::global_vertex_index> indices(n_cpu);
     int ierr = MPI_Allgather(&next_index, 1, DEAL_II_DOF_INDEX_MPI_TYPE, indices.data(),
-                             indices.size(), DEAL_II_DOF_INDEX_MPI_TYPE, triangulation.get_communicator());
+                             1, DEAL_II_DOF_INDEX_MPI_TYPE, triangulation.get_communicator());
```

Comment:
Oh, what an insidious problem!

Reply:
@bangerth bonus points if you can spot the other bug without looking at #6499 :)

I checked and all other calls to MPI_Allgather look right.

Reply:
No way. It's hard to see the difference even in the diffs on #6499.